src/app/core/services/cluster.service.spec.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ClusterService} from './cluster';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {TestBed} from '@angular/core/testing';
+import {MatDialogModule} from '@angular/material/dialog';
+import {AppConfigMockService} from '@test/services/app-config-mock';
+import {AppConfigService} from '@app/config.service';
+import {lastValueFrom} from 'rxjs';
+import {machineDeploymentsFake} from '@test/data/node';
+import {fakeDigitaloceanCluster} from '@test/data/cluster';
+import {fakeProject} from '@test/data/project';
+import {environment} from '@environments/environment';
+import {MatSnackBarModule} from '@angular/material/snack-bar';
+import {RouterTestingModule} from '@angular/router/testing';
+
+describe('ClusterService', () => {
+  let service: ClusterService;
+  let httpController: HttpTestingController;
+
+  const mockMachineDeploymentID = machineDeploymentsFake()[0].id;
+  const mockClusterID = fakeDigitaloceanCluster().id;
+  const mockProjectID = fakeProject().id;
+
+  const newRestRoot = environment.newRestRoot;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [ClusterService, {provide: AppConfigService, useClass: AppConfigMockService}],
+      imports: [HttpClientTestingModule, MatDialogModule, MatSnackBarModule, RouterTestingModule],
+    });
+    httpController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(ClusterService);
+  });
+
+  afterEach(() => {
+    httpController.verify();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should dispatch correct GET request when externalMachineDeploymentNodes() is called', () => {
```

Comment:
IMO, you don't need to write `correct` word since you are already verifying by assertion

`should dispatch GET request on externalMachineDeploymentNodes() is called` 

OR

`should dispatch GET request on externalMachineDeploymentNodes() method invoked`